core/class/jMQTTBase.class.php

```diff
@@ -176,16 +176,37 @@ protected static function send_to_mqtt_daemon($params) {
       socket_close($socket);
    }
 
-   public static function new_mqtt_client($id, $hostname, $port = 1883, $clientid = '', $statustopic = '', $username = '', $password = '') {
+   public static function new_mqtt_client($id, $hostname,       $port = 0,           $clientid = '',
+                                          $statustopic = '',    $username = '',      $password = '',
+                                          $tls = 'disable',     $tlscafile = '',     $tlssecure = '0',
+                                          $tlsclicertfile = '', $tlsclikeyfile = '', $paholog='') {
       $params['cmd']='newMqttClient';
       $params['id']=$id;
       $params['callback']='ws://127.0.0.1:'.config::byKey('websocketport', get_called_class(), get_called_class()::DEFAULT_WEBSOCKET_PORT).'/plugins/jMQTT/core/php/jmqttd.php';
       $params['hostname']=$hostname;
-      $params['port']=$port;
       $params['clientid']=$clientid;
       $params['statustopic']=$statustopic;
       $params['username']=$username;
       $params['password']=$password;
+      if ($tlsclicertfile != '')
+         $params['tlsclicertfile']=realpath(dirname(__FILE__) . '/../../data/certs/'.$tlsclicertfile);
```

Comment:
I would prefer that only absolute path is passed to this function. Path construction needs to be done in jMQTT.class.php

Reply:
Can I rework new_mqtt_client() function?
It has way too much parameters, can I pass $params directly in?

Reply:
yes

Reply:
better :-)